libs/eo/metering-points/shell/src/lib/eo-metering-points.store.ts

```diff
@@ -65,15 +57,21 @@ export class EoMeteringPointsStore extends ComponentStore<EoMeteringPointsState>
     })
   );
 
-  private readonly setCertificateContract = this.updater(
+  private readonly setContract = this.updater(
     (state, contract: EoCertificateContract): EoMeteringPointsState => ({
       ...state,
-      meteringPoints: state.meteringPoints.map((mp) => {
-        if (mp.gsrn === contract.gsrn) {
-          mp.contract = contract;
-        }
-        return mp;
-      }),
+      meteringPoints: state.meteringPoints.map((mp) =>
+        mp.gsrn === contract.gsrn ? { ...mp, contract } : mp
+      ),
+    })
+  );
+
+  private readonly setContractLoading = this.updater(
```

Comment:
Just a comment about naming of this function: I think this should be named `toggleContractLoading` as it is toggling if the contract is loading or not

Reply:
Right you are @duizer! :)